app/lib/features/events/providers/notifiers/event_notifiers.dart

```diff
@@ -0,0 +1,60 @@
+import 'dart:async';
+import 'package:acter/features/home/providers/client_providers.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart' as ffi;
+import 'package:riverpod/riverpod.dart';
+
+class EventListNotifier
+    extends FamilyAsyncNotifier<List<ffi.CalendarEvent>, String?> {
+  late Stream<bool> _listener;
+
+  Future<List<ffi.CalendarEvent>> _getEventList() async {
+    final client = ref.watch(alwaysClientProvider);
+    //GET ALL EVENTS
+    if (arg == null) {
+      return (await client.calendarEvents()).toList();
+    } else {
+      //GET SPACE EVENTS
+      final space = await client.space(arg!);
```

Comment:
let's not create another space object, but the one cached by the spaceProvider here.

```suggestion
    //GET ALL EVENTS
    if (arg == null) {
      final client = ref.watch(alwaysClientProvider);
      return (await client.calendarEvents()).toList();
    } else {
      //GET SPACE EVENTS
      final space = await ref.read(spaceProvider(arg!));
      if (space == null) return [];
```

Reply:
Did not find calendarEvent on this approach.

<img width="489" alt="Screenshot 2024-07-31 at 12 25 37 PM" src="https://github.com/user-attachments/assets/d93881a9-a247-4cc6-9405-f6f6e6a73412">